### DISCUSSION TITLE
Is there a way that one Magma network can talk to another, so subscribers can use both?

### DISCUSSION THREAD

**Discussion by ShubhamTatvamasi** [magma/magma#5499] [Q&A] [answered]
Like Roaming:
- Home Network
- Roaming Network

**Comment by hcgatewood** [comment]
cc @shaddi

**Comment by karthiksubraveti** [comment]
Can you help understand the usecase better ? Are these networks managed by different operators ?

**Reply by ShubhamTatvamasi** [reply]
Yes, different operators.

**Reply by karthiksubraveti** [reply]
As far as I know, we only allow federation through HSS. We don't have a equivalent mechanism of using federation gateway with magma local subscriberdb. @uri200 Do you have more context on this ?

**Comment by shaddi** [comment]
@uri200 can correct me if I'm wrong, but this isn't currently possible (though it is an interesting idea).

**Comment by uri200** [comment, marked as answer]
Sorry, but I think I didn't fully understand the question,.so let me know if that is of any help:

The way operator networks talk each other is through 3gpp interfaces like s8. Inbound roaming will implement that interface. Since those remote networks are 3gpp based, we have to follow the standard to communicate and proceed. There is no way around it.

In case we had two Magma networks we could think on some kind of grpc procedure to replace S8.  But see that if we implement 3gpp inbound/outbound roaming we can plug two Magma networks already using those interfaces. No need to create our own grpc protocol.

Also see there is a security component were operators will not want to share their subscribers with other operators. That is why 3gpp provides S8 interface, so remote subscribers are allowed on local networks without the need of sharing subscriber credentials.

Right now the supported mode will be: local subscribers are authenticated with subscriber db and remote with HSS.  But I already have a proposal to make that even more generic and be able to use subscriberdb and local hss for local subscribers (I call it Hybrid Federated). That requires changes in many different systems, so that will be explored once inbound roaming works.

Please, let me know if that answers. your question

**Reply by ShubhamTatvamasi** [reply]
Yes, Thanks you.
